src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -476,6 +477,30 @@ public void sendLocalNodeRequest(TransportService transportService) {
         }
     }
 
+    /**
+     * Requests the ActionListener onFailure method to be run by OpenSearch.  The result will be handled by a {@link ActionListenerOnFailureResponseHandler}.
+     *
+     * @param transportService  The TransportService defining the connection to OpenSearch.
+     * @param failureException The exception to be sent to OpenSearch
+     */
+    public void sendActionListenerOnFailureRequest(TransportService transportService, Exception failureException) {
```

Comment:
How is this Request associated with the ActionListener instantiated inside the `startActionListener()` method or anywhere else?  Where (other than the test) is this method ever called?

Reply:
It is not called anywhere yet, but that functionality will be added later when the actual createDetector is implemented as far as I know.

Reply:
We are creating these APIs for SDK like:
https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/ExtensionsRunner.java#L443

https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/ExtensionsRunner.java#L463

They will be called once we integrate SDK with AD extension.